app/js/arethusa.search/services/search.js

```diff
@@ -40,6 +40,182 @@ angular.module('arethusa.search').service('search', [
       });
     };
 
+    this.findWordInContext = function(query) {
```

Comment:
I think this function's readability would benefit from being split up into smaller functions.

Reply:
ok